packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -1241,9 +1241,11 @@ class ProjectFileInvalidator {
       final List<Future<void>> waitList = <Future<void>>[];
       for (final Uri uri in urisToScan) {
         waitList.add(pool.withResource<void>(
-          () => _fileSystem
-            .file(uri)
-            .stat()
+          // Calling fs.stat() is more performant than fs.file().stat(), but
```

Comment:
I would think these would be the same 😕

Reply:
My guess is because fs.file().stat() generated an additional object in each loop. Although, IIUC, Dart is using generational GC and this kind of short lived objects shouldn't affect the performance so much.